src/api-mocks/base.js

```diff
@@ -0,0 +1,21 @@
+import cloneDeep from 'lodash/cloneDeep';
+import set from 'lodash/set';
+
+export const BASE_URL = process.env.REACT_APP_BASE_API_URL || 'http://localhost:8000/api/v2/';
+
+export const getDefaultFactory = default_object => {
+  /**
+   * @param  {Object} overrides Key-value mapping with overrides from the defaults. These
+   *                            are deep-assigned via lodash.set to the defaults, so use
+   *                            '.'-joined strings as keys for deep paths.
+   * @return {Object}           A form detail object conforming to the Form proptype spec.
+   */
+  const defaultFactory = (overrides = {}) => {
+    const obj = cloneDeep(default_object);
+    for (const [key, value] of Object.entries(overrides)) {
+      set(obj, key, value);
+    }
+    return obj;
+  };
+  return defaultFactory;
+};
```

Comment:
ah there was no particular reason other than easy of putting a debugger before the return statement. I don't have any particular strong preferences.

The docbloc does belong to the curried function though, the outside function is a factory that only takes in a `default_object` argument & returns a function. We don't currently generate any documentation from these docblocks so it's more whatever your editor uses. Preferably we switch to TypeScript anyway so the structured typing information is in the code rather than the annotations.